### PR TITLE
more vending fix

### PR DESCRIPTION
**Pull request by Tlaltecuhtli**
stacks were weird because if you put 2 bars of the same type they will count 2 but they would merge while inside the vendor

on server for some reason vend_ready would default to 0 after 1 use ( it doesnt happen in local) but i guess putting the vend_ready = 1 at the end indented more on the left should fix it?

this branch works fine so i guess it would on server too

edit: turns out the vend_ready not going back to 1 once it finishes doing stuff was caused by the ishuman() check failing on ghosts n shit now its fix

### CODE REVIEW

File: code/modules/vending/_vending.dm [hunk 1]
```diff
@@ -853,23 +856,26 @@ GLOBAL_LIST_EMPTY(vending_products)
...
+						owner.adjust_money(S.custom_price)
+					vending_machine_input[N] = max(vending_machine_input[N] - 1, 0)
+					if(last_shopper != usr || purchase_message_cooldown < world.time)
+						say("Thank you for buying local and purchasing [S]!")
+						purchase_message_cooldown = world.time + 5 SECONDS
+						last_shopper = usr
+					S.forceMove(drop_location())
+					loaded_items--
+					use_power(5)
+		vend_ready = 1
```
Comment (AnturK) [inline code/modules/vending/_vending.dm]: One tab more or every href will reset it making it meaningless.

**Review by AnturK — CHANGES_REQUESTED** [review]

File: code/modules/vending/_vending.dm [hunk 2]
```diff
@@ -853,23 +856,34 @@ GLOBAL_LIST_EMPTY(vending_products)
...
-				use_power(5)
-				vending_machine_input[N] = max(vending_machine_input[N] - 1, 0)
-				if(last_shopper != usr || purchase_message_cooldown < world.time)
-					say("Thank you for buying local and purchasing [S]!")
-					purchase_message_cooldown = world.time + 5 SECONDS
-					last_shopper = usr
-				S.forceMove(drop_location())
-				loaded_items--
-				vend_ready = 1
-	updateUsrDialog()
```
Comment (AnturK) [inline code/modules/vending/_vending.dm]: You removed dialog update.

**Review by AnturK — CHANGES_REQUESTED** [review]

File: code/modules/vending/_vending.dm (hunk 2 above)
Comment (Tlaltecuhtli) [reply code/modules/vending/_vending.dm]: it updates when it vends the item and it doesnt when you dont have money

Comment (AnturK) [reply code/modules/vending/_vending.dm]: Alright, thought there were more topic branches above this.

**Pull request merged**